src/psyclone/transformations.py

```diff
@@ -237,52 +222,75 @@ def _validate(self, node1, node2):
                                       "next to each other")
-
-        # Check iteration space is the same
+        # Check that iteration space is the same
         if node1.iteration_space != node2.iteration_space:
             raise TransformationError("Error in LoopFuse transformation. "
                                       "Loops do not have the same "
                                       "iteration space")
 
     def apply(self, node1, node2):
-        '''Fuse the loops represented by :py:obj:`node1` and
-        :py:obj:`node2`.'''
+        ''' Fuses two Loops represented by `psyclone.psyGen.Node` objects
+        after performing validity checks.
+
+        :param node1: the first Node we are checking
+        :type node1: :py:class:`psyclone.psyGen.Node`
+        :param node2: the second Node we are checking
+        :type node2: :py:class:`psyclone.psyGen.Node`
+        :returns: two-tuple of the modified Schedule and a record of \
+                  the transformation.
+        :rtype: (:py:class:`psyclone.psyGen.Schedule`, \
+                 :py:class:`psyclone.undoredo.Memento`).
+        '''
 
+        # Validity checks for the supplied nodes
         self._validate(node1, node2)
 
         schedule = node1.root
 
-        # create a memento of the schedule and the proposed transformation
+        # Create a memento of the schedule and the proposed transformation
         keep = Memento(schedule, self, [node1, node2])
 
-        # add loop contents of node2 to node1
+        # Add loop contents of node2 to node1
         node1.loop_body.children.extend(node2.loop_body)
 
-        # change the parent of the loop contents of node2 to node1
+        # Change the parent of the loop contents of node2 to node1
         for child in node2.loop_body:
             child.parent = node1.loop_body
 
-        # remove node2
+        # Remove node2
         node2.parent.children.remove(node2)
 
         return schedule, keep
 
 
 class GOceanLoopFuseTrans(LoopFuseTrans):
-    ''' Performs error checking (that the loops are over the same grid-point
-        type) before calling the :py:meth:`LoopFuseTrans.apply` method of the
-        :py:class:`base class <LoopFuseTrans>` in order to fuse two
-        GOcean loops. '''
+    ''' GOcean API application of the :py:class:`base class <LoopFuseTrans>`
+    in order to fuse two GOcean Loops after performing validity checks (e.g.
+    that the loops are over the same grid-point type). For example:
+
+    >>> from psyclone.parse.algorithm import parse
+    >>> from psyclone.psyGen import PSyFactory
+    >>> ast, invokeInfo = parse("shallow_alg.f90")
+    >>> psy = PSyFactory("gocean1.0").create(invokeInfo)
+    >>> schedule = psy.invokes.get('invoke_0').schedule
+    >>> schedule.view()
+    >>>
+    >>> from psyclone.transformations import GOceanLoopFuseTrans
+    >>> ftrans = GOceanLoopFuseTrans()
+    >>> new_schedule, memento = ftrans.apply(schedule.children[0],
```

Comment:
I didn't see this level of `schedule.children` changed neither in Joerg's #454 (simplify walk) nor in Sergi's #400 for `DO` loops and I updated the branch with the master after those two PRs were merged.

Reply:
I presume `__getitem__` in `Schedule` instead of `get_item`? I modified this but I did not modify the related transformation scripts in `examples/gocean` to this way of applying the transformation because there are quite a lot of changes already. Plus, there are a couple more scripts that apply `LoopFuseTrans` in `dynamo/eg2` which should then also be modified this way. I am happy to update the scripts in this issue or leave it for another one.